File: packages/android/formbricksSDK/src/main/java/com/formbricks/formbrickssdk/model/environment/Project.kt
package com.formbricks.formbrickssdk.model.environment

import com.google.gson.annotations.SerializedName
import kotlinx.serialization.Serializable

@Serializable
data class Project(
    @SerializedName("id") val id: String?,
    @SerializedName("recontactDays") val recontactDays: Double?,
    @SerializedName("clickOutsideClose") val clickOutsideClose: Boolean?,
    @SerializedName("darkOverlay") val darkOverlay: Boolean?,
    @SerializedName("placement") val placement: String?,
    @SerializedName("inAppSurveyBranding") val inAppSurveyBranding: Boolean?,
    @SerializedName("styling") val styling: Styling?
)

File: apps/web/app/api/v1/client/[environmentId]/environment/lib/project.ts
import { cache } from "@/lib/cache";
import { projectCache } from "@/lib/project/cache";
import { validateInputs } from "@/lib/utils/validate";
import { Prisma } from "@prisma/client";
import { cache as reactCache } from "react";
import { prisma } from "@formbricks/database";
import { logger } from "@formbricks/logger";
import { ZId } from "@formbricks/types/common";
import { DatabaseError } from "@formbricks/types/errors";
import { TJsEnvironmentStateProject } from "@formbricks/types/js";

export const getProjectForEnvironmentState = reactCache(
  async (environmentId: string): Promise<TJsEnvironmentStateProject | null> =>
    // [UseTusk]

    cache(
      async () => {
        validateInputs([environmentId, ZId]);

        try {
          return await prisma.project.findFirst({
            where: {
              environments: {
                some: {
                  id: environmentId,
                },
              },
            },
            select: {
              id: true,
              recontactDays: true,
              clickOutsideClose: true,
              darkOverlay: true,
              placement: true,
              inAppSurveyBranding: true,
              styling: true,
            },
          });
        } catch (error) {
          if (error instanceof Prisma.PrismaClientKnownRequestError) {
            logger.error(error, "Error getting project for environment state");
            throw new DatabaseError(error.message);
          }
          throw error;
        }
      },
      [`getProjectForEnvironmentState-${environmentId}`],
      {
        tags: [projectCache.tag.byEnvironmentId(environmentId)],
      }
    )()
);
